modules/aws/README.md

```diff
@@ -8,3 +8,51 @@ This directory contains all the AWS Terraform modules.
 - [`eks-global`](eks-global/README.md)
 - [`eks`](eks/README.md)
 - [`irsa`](irsa/README.md)
+
+## Style Guide
+
+This section covers AWS specific standards that all modules shoudl try to follow to decrease the risk of annoying problems in the future, and to guide
+new contributors in how resources should be named. Any changes to this guide should be done together with updating the Terraform.
+
+
+### Names and tags
+A quirk of AWS is that there are some resources that are global and other that are regional. Ontop of that there are resources that get automatically
+generated IDs and others that use the given name as a unique ID. That name may either have to be unique within the account, within that AWS region or
+across all AWS regions.
+
+Global resources with unique names:
+* IAM Roles
+* IAM Policy
+
+Regional resources with unique names:
+* EKS Clusters
+* S3 Bucket
+
+Any resource that has a generated ID should have tags to distinguish the resources between each other. The following tags should be added to all
+resources of this kind. If multiple instances of the same resource is expected to be deployed, such as EKS, the name of the resource should include a
+unique suffix to distinguish the resources.
+| Tag | Required | Description |
+| --- | --- | --- |
+| Name | Yes | Name of the resource, try to convey the purpose or use. |
+| Environment | Yes | Environment the resource is deployed in. |
+| Module | Yes | The name of the Terraform module which has created the resource. |
```

Comment:
Is this not rather the module that defined the resources? It may well because of configuration in a downstream module.

Reply:
We should not have nested modules except in very specific circumstances. The only nested module we have today is IRSA which is according to me is useful. I think it is easiest to have the first module name in the tag.